### PR TITLE
Replace call to BINARY with CAST(_ AS BINARY)

### PR DESCRIPTION
**Pull request by matt-mcmurray-shopify**
### Motivation / Background
We started experiencing the following error when running our test suite:
```
Minitest::UnexpectedError: MysqlWarning: 'BINARY expr' is deprecated and will be removed in a future release. Please use CAST instead
```

This can be confirmed in the MySQL developer docs: https://dev.mysql.com/doc/refman/8.0/en/cast-functions.html#operator_binary

This Pull Request has been created because if not updated before `BINARY` is deprecated, ActiveRecord will be broken for MySQL.

### Detail

This Pull Request changes the call to `BINARY` with a call to `CAST( <EXPRESSION> AS BINARY)`. As recommended in the warning/MySQL docs.

### Checklist

Before submitting the PR make sure the following are checked:

* [X] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [X] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: activerecord/lib/arel/visitors/mysql.rb [hunk 1]
```diff
@@ -5,8 +5,9 @@ module Visitors
     class MySQL < Arel::Visitors::ToSql
       private
         def visit_Arel_Nodes_Bin(o, collector)
-          collector << "BINARY "
+          collector << "CAST( "
```
Comment (rafaelfranca) [inline activerecord/lib/arel/visitors/mysql.rb:8]:
```suggestion
          collector << "CAST("
```

**Review by rafaelfranca — COMMENTED** [review]
Tests are broken after this change. Can you take a look?

**Comment by skipkayhil** [comment]
Related, should we even be testing `Arel::Visitors::MySQL` with other databases:
https://buildkite.com/rails/rails/builds/94256#018694c4-8558-43d7-91f6-59c97ffd83cd/1016-1029

**Comment by rafaelfranca** [comment]
It is fine since it isn't testing the database itself. But maybe we should run the arel test suite in a different task only once, no need to run 4 times.

**Comment by yahonda** [comment]
I think it needs some tests to validate why it is needed.

File: activerecord/lib/arel/visitors/mysql.rb (hunk 1 above)
Comment (matt-mcmurray-shopify) [reply activerecord/lib/arel/visitors/mysql.rb:8]: Yep, I'll address these 👍

**Comment by matt-mcmurray-shopify** [comment]
@yahonda I've updated the existing test to assert for the new `CAST(...)` syntax, did you want any additional test cases?

**Pull request merged**